admin/modules/forum/attachments.php

```diff
@@ -858,14 +858,18 @@ function() {
 			if($num_results > $mybb->input['perpage'])
 			{
 				$pagination_url = "index.php?module=forum-attachments&amp;results=1";
-				$pagination_vars = array('perpage', 'sortby', 'order', 'filename', 'mimetype', 'username', 'fid', 'downloads', 'downloads_dir', 'dateuploaded', 'dateuploaded_dir', 'filesize', 'filesize_dir');
+				$pagination_vars = array('perpage', 'sortby', 'order', 'filename', 'mimetype', 'username', 'downloads', 'downloads_dir', 'dateuploaded', 'dateuploaded_dir', 'filesize', 'filesize_dir');
 				foreach($pagination_vars as $var)
 				{
 					if($mybb->input[$var])
 					{
 						$pagination_url .= "&{$var}=".urlencode($mybb->input[$var]);
 					}
 				}
+				if($mybb->input['forum'])
+				{
+					$pagination_url .= "&forum=".implode("|", $mybb->input['forum']);
```

Comment:
This might not make sense out of context, but if you change the other line to expect `$mybb->input['forum']` to be an array of `fid`s, then you can just set it up here like this:

```suggestion
					$pagination_url .= "&forum[]=".implode("&forum[]=", $mybb->input['forum']);
```

The intended, end result for passing arrays over `GET` is `&var[]=1&var[]=2&var[]=3` and that yields `[1,2,3]`